inception-imls-dl4j/src/main/java/de/tudarmstadt/ukp/inception/recommendation/imls/dl4j/pos/DL4JSequenceRecommender.java

```diff
@@ -263,14 +263,22 @@ private DataSet vectorize(List<? extends Sample> aData, Object2IntMap<String> aT
     
                 featureVec.put(new INDArrayIndex[] { point(sampleIdx), all(), point(t) }, vector);
                 featureMask.putScalar(new int[] { sampleIdx, t }, 1.0);
-                labelMask.putScalar(new int[] { sampleIdx, t }, 1.0);
-    
-                if (aIncludeLabels) {
+
+                // exclude padding labels from training
+                // compare instances to avoid collision with possible no_label user label
+                if (labels != null && labels.get(t) != NO_LABEL) {
+                    labelMask.putScalar(new int[] { sampleIdx, t }, 1.0);
+                }
+
+                if (aIncludeLabels && labels != null) {
                     String label = labels.get(t);
-                    if (!aTagset.containsKey(label)) {
-                        aTagset.put(label, aTagset.size());
+                    // do not add padding label no_label as predictable label
+                    if (label != NO_LABEL) {
+                        if (!aTagset.containsKey(label)) {
```

Comment:
Was like this before but better use `Map.computeIfAbsent()` which performs this as an atomic operation.